core/elite_edge_node.go

```diff
@@ -107,17 +105,44 @@ func (a *AggregatedEENVotes) Merge(b *AggregatedEENVotes) (*AggregatedEENVotes,
 	if a.Block != b.Block {
 		return nil, errors.New("Cannot merge incompatible votes")
 	}
-	newMultiplies := make([]uint32, len(a.Multiplies))
+	newMultiplies := []uint32{}
+	newAddresses := []common.Address{}
+
 	isSubset := true
-	for i := 0; i < len(a.Multiplies); i++ {
-		newMultiplies[i] = a.Multiplies[i] + b.Multiplies[i]
-		if newMultiplies[i] < a.Multiplies[i] || newMultiplies[i] < b.Multiplies[i] {
-			return nil, errors.New("Signiature multipliers overflowed")
-		}
-		if a.Multiplies[i] == 0 && b.Multiplies[i] != 0 {
+	i := 0
+	j := 0
+	for i < len(a.Addresses) && j < len(b.Addresses) {
+		cmp := bytes.Compare(a.Addresses[i].Bytes(), b.Addresses[j].Bytes())
+		if cmp == 0 {
+			sum := a.Multiplies[i] + b.Multiplies[j]
+			if sum < a.Multiplies[i] || sum < b.Multiplies[i] {
+				return nil, errors.New("Signiature multipliers overflowed")
+			}
+			newMultiplies = append(newMultiplies, sum)
```

Comment:
Need the following line after Line 121?

newAddresses = append(newAddresses, a.Addresses[i])

I also added a new testcase "TestEliteAggregateVoteMerge" in elite_edg_node_test.go for this merging process

Reply:
Fixed it, will commit shortly